pkg/login/client_login.go

```diff
@@ -155,6 +156,51 @@ func GuestLogin(config *config.Config) (string, error) {
 	return guest_user.Url, nil
 }
 
+func CILogin(config *config.Config, apiKey string, name string) error {
+	parsedBaseURL, err := url.Parse(config.APIBaseURL)
+	if err != nil {
+		return err
+	}
+
+	client := &hookdeck.Client{
+		BaseURL: parsedBaseURL,
+		APIKey: apiKey,
+	}
+
+	deviceName := name
+	if deviceName == "" {
+		deviceName = config.Profile.DeviceName
+	}
+	response, err := client.CreateCIClient(hookdeck.CreateCIClientInput{
+		DeviceName: deviceName,
+	})
+	if err != nil {
+		return err
+	}
+
+	validateErr := validators.APIKey(response.APIKey)
```

Comment:
nitpicks: You don't need to assign a new variable here, you can reuse `err`. `validators.APIKey` should return `nil` if there is no error.

Also, when the return value of a method is only an error value,

you can write the validation like so:

```
if err := validators.APIKey(response.APIKey); err != nil {
  return err
}
```

Reply:
Awesome, TIL! Pushed the changes with your feedback